plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/experimental/data/MapboxOverviewDataSource.kt

```diff
@@ -0,0 +1,15 @@
+package com.mapbox.maps.plugin.viewport.experimental.data
+
+class MapboxOverviewDataSource : ViewportDataSource {
```

Comment:
This class and `MapboxFollowingDataSource` are almost the same, can you integrate them into one class with a `type` property?

Reply:
The implementation would be quite different if user want to customise their own ViewportState and provide their own data source. I intentionally decoupled the overview and following mode so that we user can provide their own mode and data source using the same interface we used internally for following and overview states.